packages/mask/src/components/InjectedComponents/SetupGuide.tsx

```diff
@@ -477,31 +413,54 @@ function SetupGuideUI(props: SetupGuideUIProps) {
     )
     //#endregion
 
+    const { value: persona_ } = useAsync(async () => {
+        return Services.Identity.queryPersona(Identifier.fromString(persona.toText(), ECKeyIdentifier).unwrap())
+    }, [persona])
+
     const onConnect = async () => {
         // attach persona with SNS profile
         await Services.Identity.attachProfile(new ProfileIdentifier(ui.networkIdentifier, username), persona, {
             connectionConfirmState: 'confirmed',
         })
 
         // auto-finish the setup process
-        const persona_ = await Services.Identity.queryPersona(
-            Identifier.fromString(persona.toText(), ECKeyIdentifier).unwrap(),
-        )
-        if (!persona_.hasPrivateKey) throw new Error('invalid persona')
-        await Services.Identity.setupPersona(persona_.identifier)
+        if (!persona_?.hasPrivateKey) throw new Error('invalid persona')
+        await Services.Identity.setupPersona(persona_?.identifier)
         MaskMessages.events.ownPersonaChanged.sendToAll(undefined)
     }
 
     const onClose = () => {
         currentSetupGuideStatus[ui.networkIdentifier].value = ''
     }
 
+    const onDone = () => {
+        // check user guide status
+        const network = ui.networkIdentifier
+        if (network === 'twitter.com' && userGuideStatus[network].value !== 'completed') {
```

Comment:
what about other network?

Reply:
user guide is designed for twitter network currently.